### PR TITLE
fix(sdk): custom bind method on awscdk targets

### PR DESCRIPTION
**Pull request by marciocadev**
Following @eladb's recommendation in PR #3398, I removed the import of `target-awscdk` and `target-tf-aws`, and checked if the `host` implements the methods of the `IAwsFunction` interface. This way, the error that was occurring due to not having `aws-cdk-lib` globally installed no longer happens.

## Checklist

- [x] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [x] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

File: examples/tests/valid/dynamo_awscdk.w [hunk 1]
```diff
@@ -0,0 +1,96 @@
+/*\
+skip: true
```
Comment (eladb) [inline examples/tests/valid/dynamo_awscdk.w:2]: @tsuf98 is this still going to run in our integration tests on aws?

File: examples/tests/valid/dynamo_awscdk.w [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      billingMode: awscdk.aws_dynamodb.BillingMode.PAY_PER_REQUEST,
+      partitionKey: awscdk.aws_dynamodb.Attribute {
+        name: "Flavor",
+        type: awscdk.aws_dynamodb.AttributeType.STRING,
+      },
+    );
+    this.tableName = this.table.tableName;
+  }
+
+  _bind(host: std.IInflightHost, ops: Array<str>) {
```
Comment (eladb) [inline examples/tests/valid/dynamo_awscdk.w:42]: @Chriscbr it feels weird that people need to override an internal method... maybe we should formalize it? (At least change it to `bind()`.

(Would be nice to have something like a "private protected" method which can only be overridden by derived classes but not called from outside)

File: examples/tests/valid/dynamo_awscdk.w [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+let table = new DynamoTable();
+
+test "put an item in the table" {
+  table.putItem({
+    "Flavor" => Attribute {
+      type: AttributeType.String,
+      value: "Chocolate",
+    },
+  });
```
Comment (eladb) [inline examples/tests/valid/dynamo_awscdk.w:95]: No assertion in the test

File: libs/wingsdk/src/shared-aws/function.ts [hunk 4]
```diff
@@ -30,10 +29,18 @@ export class Function {
...
   public static from(host: IInflightHost): IAwsFunction | undefined {
-    if (host instanceof TfAwsFunction) {
+    if (this.checkIAwsFunctionImplementation(host)) {
       return host;
     }
-
     return undefined;
   }
+
+  private static checkIAwsFunctionImplementation(
```
Comment (eladb) [inline libs/wingsdk/src/shared-aws/function.ts:38]:
```suggestion
  private static isAwsFunction(
```

File: examples/tests/valid/dynamo_awscdk.w (hunk 1 above)
Comment (tsuf239) [reply examples/tests/valid/dynamo_awscdk.w:2]: Only when running tests through hangar, we don't have it built-in in the CLI

File: examples/tests/valid/dynamo_awscdk.w (hunk 2 above)
Comment (Chriscbr) [reply examples/tests/valid/dynamo_awscdk.w:43]: I agree. Let's make it public

Comment (marciocadev) [reply examples/tests/valid/dynamo_awscdk.w:43]: Do you want me to make the changes in this PR, or should I open another one?

Comment (Chriscbr) [reply examples/tests/valid/dynamo_awscdk.w:43]: In another PR is fine 👍

Comment (marciocadev) [reply examples/tests/valid/dynamo_awscdk.w:43]: Open a issue #3775 and a PR #3776

**Review by Chriscbr — APPROVED** [review]

**Pull request merged**